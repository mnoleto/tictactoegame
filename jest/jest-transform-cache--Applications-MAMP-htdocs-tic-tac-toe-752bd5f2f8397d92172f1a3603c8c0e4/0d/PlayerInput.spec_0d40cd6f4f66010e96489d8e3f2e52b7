/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_PlayerInput = require('./PlayerInput'); /* istanbul ignore next */var _PlayerInput2 = _interopRequireDefault(_PlayerInput);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// PlayerInput is responsible for render the game grid
describe('PlayerInput', function () {
  var props = /* istanbul ignore next */void 0,mountedPlayerInput = /* istanbul ignore next */void 0;
  var playerInput = function playerInput() {
    if (!mountedPlayerInput) {
      mountedPlayerInput = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayerInput2.default, props));

    }
    return mountedPlayerInput;
  };

  beforeEach(function () {
    props = {
      label: 'X',
      onInputChange: jest.fn(),
      player: 'playerX',
      result: { status: 'waiting' },
      turn: 'X',
      value: 'Player 1' };

    mountedPlayerInput = undefined;
  });

  it('should render the PlayerInput', function () {
    var playerInputWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayerInput2.default, props));
    expect(playerInputWrapper).toMatchSnapshot();
  });

  it('should render a div as root element', function () {
    var playerInputElement = playerInput().find('div');
    var wrapping = playerInputElement.first();
    expect(wrapping.children()).toEqual(playerInput().children());
  });

  it('should render a label', function () {
    var playerInputElement = playerInput().find('label');
    expect(wrapping.length).toEqual(1);
  });

  describe('css classes', function () {
    it('should render a default class player-input', function () {

    });

    it('should render an active class when its his turn', function () {

    });

    it('should render an active class when its not his turn', function () {

    });
  });

  describe('input type=text', function () {
    it('should render a input type=text', function () {

    });

    it('should change its value when change props', function () {

    });

    it('simulate change event', function () {

    });

    it('should be disabled when the status is running', function () {

    });
  });
});