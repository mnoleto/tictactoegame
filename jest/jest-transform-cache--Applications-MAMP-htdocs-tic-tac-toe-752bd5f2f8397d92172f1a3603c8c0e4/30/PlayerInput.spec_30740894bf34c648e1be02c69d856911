/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_PlayerInput = require('./PlayerInput'); /* istanbul ignore next */var _PlayerInput2 = _interopRequireDefault(_PlayerInput);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// PlayerInput is responsible for render the game grid
// CONTRACTS
// 1. 

describe('PlayerInput', function () {
  var props = /* istanbul ignore next */void 0,mountedPlayerInput = /* istanbul ignore next */void 0;
  var playerInput = function playerInput() {
    if (!mountedPlayerInput) {
      mountedPlayerInput = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayerInput2.default, props));

    }
    return mountedPlayerInput;
  };

  beforeEach(function () {
    props = {
      label: 'X' };

    mountedPlayerInput = undefined;
  });

  it('always render the PlayerInput', function () {
    var playerInputElement = playerInput().find('div');
    expect(playerInputElement.length).toEqual(1);
  });
});