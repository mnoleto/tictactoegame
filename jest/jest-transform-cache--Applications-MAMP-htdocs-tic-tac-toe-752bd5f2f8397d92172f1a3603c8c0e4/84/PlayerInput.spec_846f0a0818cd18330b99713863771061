/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_PlayerInput = require('./PlayerInput'); /* istanbul ignore next */var _PlayerInput2 = _interopRequireDefault(_PlayerInput);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// PlayerInput is responsible for render the game grid
// CONTRACTS
// 1. 

describe('PlayerInput', function () {
  var props = /* istanbul ignore next */void 0,mountedPlayerInput = /* istanbul ignore next */void 0;
  var playerInput = function playerInput() {
    if (!mountedPlayerInput) {
      mountedPlayerInput = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayerInput2.default, props));

    }
    return mountedPlayerInput;
  };

  beforeEach(function () {
    props = {
      label: 'X' };

    mountedPlayerInput = undefined;
  });

  it('render the PlayerInput', function () {
    var playerInputWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayerInput2.default, /* istanbul ignore next */{ label: 'x' }));
    expect(playerInputWrapper).toMatchSnapshot();
  });

  it('render a div as root element', function () {
    var playerInputElement = playerInput().find('div');
    var wrappingDiv = playerInputElement.first();
    expect(wrappingDiv.children()).toEqual(playerInput().children());
  });
});