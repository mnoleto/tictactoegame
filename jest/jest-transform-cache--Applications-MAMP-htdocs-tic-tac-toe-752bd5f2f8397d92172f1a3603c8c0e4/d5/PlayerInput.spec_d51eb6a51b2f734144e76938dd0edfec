/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_PlayerInput = require('./PlayerInput'); /* istanbul ignore next */var _PlayerInput2 = _interopRequireDefault(_PlayerInput);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// PlayerInput is responsible for render the game grid
describe('PlayerInput', function () {
  var props = /* istanbul ignore next */void 0,mountedPlayerInput = /* istanbul ignore next */void 0;
  var playerInput = function playerInput() {
    if (!mountedPlayerInput) {
      mountedPlayerInput = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayerInput2.default, props));

    }
    return mountedPlayerInput;
  };

  beforeEach(function () {
    props = {
      label: 'X',
      onInputChange: jest.fn(),
      player: 'playerX',
      result: { status: 'waiting' },
      turn: '',
      value: '' };

    mountedPlayerInput = undefined;
  });

  it('should render the PlayerInput', function () {
    var playerInputWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_PlayerInput2.default, props));
    expect(playerInputWrapper).toMatchSnapshot();
  });

  it('div contains everything else that gets rendered', function () {
    var playerInputElement = playerInput().find('div');
    var wrapping = playerInputElement.first();
    expect(wrapping.children()).toEqual(playerInput().children());
  });

  it('should render a label', function () {
    var playerInputElement = playerInput().find('label');
    expect(playerInputElement.length).toEqual(1);
  });

  it('displays the label prop as label text', function () {
    var playerInputElement = playerInput().find('label');
    expect(playerInputElement.text()).toEqual(props.label);
  });

  describe('css classes', function () {
    it('should render a default class player-input', function () {
      var playerInputElement = playerInput().find('div');
      var wrapping = playerInputElement.first();
      expect(wrapping.hasClass('player-input')).toBeTruthy();
    });

    it('should render an active class when its his turn', function () {
      var inputRenderer = playerInput();
      var wrapping = playerInput().find('div').first();
      inputRenderer.setProps({ turn: 'X' });
      expect(inputRenderer.state().class).toEqual('active');
      expect(wrapping.hasClass('active')).toBeTruthy();
      expect(wrapping.hasClass('inactive')).toBeFalsy();
    });

    it('should render an active class when its not his turn', function () {
      var inputRenderer = playerInput();
      var wrapping = playerInput().find('div').first();
      inputRenderer.setProps({ turn: 'O' });
      expect(inputRenderer.state().class).toEqual('inactive');
      expect(wrapping.hasClass('inactive')).toBeTruthy();
      expect(wrapping.hasClass('active')).toBeFalsy();
    });
  });

  describe('input type=text', function () {
    it('should render a input type=text', function () {
      var playerInputElement = playerInput().find('input');
      expect(playerInputElement.length).toEqual(1);
    });

    it('should change its value when change props', function () {
      var inputRenderer = playerInput();
      var inputElement = playerInput().find('input').first();

      inputRenderer.setProps({ value: 'Player 1' });
      expect(inputElement.props().value).toEqual('Player 1');

      inputRenderer.setProps({ value: 'Player 2' });
      expect(inputElement.props().value).toEqual('Player 2');
    });

    it('simulate change event', function () {

    });

    it('should be disabled when the status is running', function () {
      var inputRenderer = playerInput();
      var inputElement = playerInput().find('input').first();
      inputRenderer.setProps({ result: { status: 'running' } });
      expect(inputRenderer.state().disabled).toBeTruthy();
      expect(inputElement.props().disabled).toBeTruthy();
    });

    it('should be enabled when the status is not running', function () {
      var inputRenderer = playerInput();
      var inputElement = playerInput().find('input').first();
      inputRenderer.setProps({ result: { status: 'waiting' } });
      expect(inputRenderer.state().disabled).toBeFalsy();
      expect(inputElement.props().disabled).toBeFalsy();
    });
  });
});