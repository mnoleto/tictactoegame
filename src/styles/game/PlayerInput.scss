@import '../_colors.scss';

$input-height: 50px;

.player-input {
  display: block;
  margin: 0px 5px 10px;
  // width: 50%;
  label {
    background-color: #fff;
    border-radius: 4px 0px 0px 4px;
    color: $blue;
    display: inline-block;
    height: $input-height;
    font-family: 'Quicksand', sans-serif;
    font-size: 200%;
    font-weight: 700;
    line-height: $input-height;
    padding: 0px 10px;
    width: 15%;
  }
  input[type="text"] {
    border: 0px;
    border-radius: 0px 4px 4px 0px;
    color: $dark-blue;
    display: inline-block;
    font-size: 200%;
    height: $input-height;
    width: 85%;
  }
  &.inactive label, &.inactive input {
    opacity: 0.5;
  }
  @media (min-width: 992px) {
    display: inline-block;
    width: auto;
  }
}