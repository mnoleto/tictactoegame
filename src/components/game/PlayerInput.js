import React, { Component, PropTypes } from 'react';
import '../../styles/game/PlayerInput.scss';

class PlayerInput extends Component {
  constructor(props) {
    super(props);

    this.state = {
      class: '',
      disabled: false
    };

    this.handleInputChange = this.handleInputChange.bind(this);
    this.isButtonDisabled = this.isButtonDisabled.bind(this);
    this.setClass = this.setClass.bind(this);
  }

  componentDidUpdate(prevProps) {
    if(this.props.turn !== '' && prevProps.turn !== this.props.turn) {
      this.setClass();
    }
    if(this.props.result.status && prevProps.result.status !== this.props.result.status) {
      this.isButtonDisabled();
    }
  }

  handleInputChange(event) {
    const { onInputChange, player } = this.props;
    onInputChange(player, event.target.value);
  }

  isButtonDisabled() {
    const { result } = this.props;
    if(result.status && (result.status === 'running' || result.status === 'finished')) {
      this.setState({disabled: true});
    } else {
      this.setState({disabled: false});
    }
  }

  setClass() {
    const { label, turn } = this.props;
    if(label === 'O') {
      this.setState({class: turn === 'O' ? 'active': (turn === 'X' ? 'inactive' : '')});
    } else {
      this.setState({class: turn === 'X' ? 'active': (turn === 'O' ? 'inactive' : '')});
    }
  }

  render() {
    const { label, player, value } = this.props;

    return (
      <div className={`player-input ${this.state.class}`}>
        <label>{label}</label>
        <input
          type="text"
          disabled={this.state.disabled}
          name={player}
          placeholder="Player name"
          value={value || ''}
          onChange={this.handleInputChange} />
      </div>
    );
  }
}

PlayerInput.propTypes = {
  label: PropTypes.string,
  onInputChange: PropTypes.func,
  player: PropTypes.string,
  result: PropTypes.object,
  turn: PropTypes.string,
  value: PropTypes.string
};

export default PlayerInput;