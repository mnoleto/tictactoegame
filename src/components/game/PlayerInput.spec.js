import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';
import PlayerInput from './PlayerInput';

// PlayerInput is responsible for render the game grid
describe('PlayerInput', () => {
  let props, mountedPlayerInput;
  const playerInput = () => {
    if(!mountedPlayerInput) {
      mountedPlayerInput = mount(
        <PlayerInput {...props} />
      );
    }
    return mountedPlayerInput;
  };

  beforeEach(() => {
    props = {
      label: 'X',
      onInputChange: jest.fn(),
      player: 'playerX',
      result: {status: 'waiting'},
      turn: '',
      value: ''
    };
    mountedPlayerInput = undefined;
  });

  it('should render the PlayerInput', () => {
    let playerInputWrapper = shallow(<PlayerInput {...props} />);
    expect(playerInputWrapper).toMatchSnapshot();
  });

  it('div contains everything else that gets rendered', () => {
    let playerInputElement = playerInput().find('div');
    let wrapping = playerInputElement.first();
    expect(wrapping.children()).toEqual(playerInput().children());
  });

  it('should render a label', () => {
    let playerInputElement = playerInput().find('label');
    expect(playerInputElement.length).toEqual(1);
  });

  it('should displays the label prop as label text', () => {
    let playerInputElement = playerInput().find('label');
    expect(playerInputElement.text()).toEqual(props.label);
  });

  describe('css classes', () => {
    it('should render a default class player-input', () => {
      let playerInputElement = playerInput().find('div');
      let wrapping = playerInputElement.first();
      expect(wrapping.hasClass('player-input')).toBeTruthy();
    });

    it('should render an active class when its his turn', () => {
      let inputRenderer = playerInput();
      let wrapping = playerInput().find('div').first();
      inputRenderer.setProps({turn: 'X'});
      expect(inputRenderer.state().class).toEqual('active');
      expect(wrapping.hasClass('active')).toBeTruthy();
      expect(wrapping.hasClass('inactive')).toBeFalsy();
    });

    it('should render an active class when its not his turn', () => {
      let inputRenderer = playerInput();
      let wrapping = playerInput().find('div').first();
      inputRenderer.setProps({turn: 'O'});
      expect(inputRenderer.state().class).toEqual('inactive');
      expect(wrapping.hasClass('inactive')).toBeTruthy();
      expect(wrapping.hasClass('active')).toBeFalsy();
    });
  });

  describe('input type=text', () => {
    it('should render a input type=text', () => {
      let playerInputElement = playerInput().find('input');
      expect(playerInputElement.length).toEqual(1);
    });

    it('should change its value when change props', () => {
      let inputRenderer = playerInput();
      let inputElement = playerInput().find('input').first();

      inputRenderer.setProps({value: 'Player 1'});
      expect(inputElement.props().value).toEqual('Player 1');

      inputRenderer.setProps({value: 'Player 2'});
      expect(inputElement.props().value).toEqual('Player 2');
    });

    it('simulate change event', () => {
      props.onInputChange = sinon.spy();
      let inputElement = playerInput().find('input').first();

      inputElement.simulate('change');
      expect(props.onInputChange.callCount).toEqual(1);
    });

    it('should be disabled when the status is running', () => {
      let inputRenderer = playerInput();
      let inputElement = playerInput().find('input').first();
      inputRenderer.setProps({result: {status: 'running'}});
      expect(inputRenderer.state().disabled).toBeTruthy();
      expect(inputElement.props().disabled).toBeTruthy();
    });

    it('should be enabled when the status is not running', () => {
      let inputRenderer = playerInput();
      let inputElement = playerInput().find('input').first();
      inputRenderer.setProps({result: {status: 'waiting'}});
      expect(inputRenderer.state().disabled).toBeFalsy();
      expect(inputElement.props().disabled).toBeFalsy();
    });
  });
});