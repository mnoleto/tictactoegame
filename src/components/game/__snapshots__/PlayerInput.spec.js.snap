exports[`PlayerInput should render the PlayerInput 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="player-input ">
    <label>
        X
    </label>
    <input
        disabled={false}
        name="playerX"
        onChange={[Function]}
        placeholder="Player name"
        type="text"
        value="" />
</div>,
  "nodes": Array [
    <div
      className="player-input ">
      <label>
            X
      </label>
      <input
            disabled={false}
            name="playerX"
            onChange={[Function]}
            placeholder="Player name"
            type="text"
            value="" />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <PlayerInput
        label="X"
        onInputChange={[Function]}
        player="playerX"
        result={
                Object {
                        "status": "waiting",
                      }
        }
        turn=""
        value="" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": PlayerInput {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleInputChange": [Function],
        "isButtonDisabled": [Function],
        "props": Object {
          "label": "X",
          "onInputChange": [Function],
          "player": "playerX",
          "result": Object {
            "status": "waiting",
          },
          "turn": "",
          "value": "",
        },
        "refs": Object {},
        "setClass": [Function],
        "state": Object {
          "class": "",
          "disabled": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="player-input ">
          <label>
                    X
          </label>
          <input
                    disabled={false}
                    name="playerX"
                    onChange={[Function]}
                    placeholder="Player name"
                    type="text"
                    value="" />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="player-input ">
          <label>
                    X
          </label>
          <input
                    disabled={false}
                    name="playerX"
                    onChange={[Function]}
                    placeholder="Player name"
                    type="text"
                    value="" />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PlayerInput
    label="X"
    onInputChange={[Function]}
    player="playerX"
    result={
        Object {
            "status": "waiting",
          }
    }
    turn=""
    value="" />,
}
`;
